eip_operator/src/controller/pod.rs

```diff
@@ -82,10 +81,10 @@ impl k8s_controller::Context for Context {
         if eip_description.network_interface_id != Some(eni_id.to_owned())
             || eip_description.private_ip_address != Some(pod_ip.to_owned())
         {
+            crate::eip::set_status_attached(&eip_api, &eip, &eni_id, pod_ip, name).await?;
             crate::aws::associate_eip(&self.ec2_client, allocation_id, &eni_id, pod_ip).await?;
+            add_dns_target_annotation(&pod_api, name, &public_ip, allocation_id).await?;
```

Comment:
in general, kubernetes operators work better when you always apply the expected state to the cluster, rather than only applying when you notice a change. is there a specific reason you wanted to move set_status_attached and add_dns_target_annotation inside the conditional here?

Reply:
ok, yeah I think this can work... I think this is to protect against multiple EIPs matching a given pod. I'm going to go back to that just being disallowed.